.github/workflows/auto-generated-checker.yml

```diff
@@ -0,0 +1,46 @@
+name: Auto-Generated Code Checker
+on:
+  push:
+    tags:
+      - "*"
+    branches:
+      - master
+      - dev
+  pull_request:
+
+jobs:
+  protobuf_codegen:
+    name: Protobuf
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: '~1.19.12'
+          check-latest: true
+      - uses: bufbuild/buf-setup-action@v1.23.1
+      - name: Installing protoc-gen-go
+        run: |
+          go install google.golang.org/protobuf/cmd/protoc-gen-go@v1.30.0
+          go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@v1.3.0
+      - shell: bash
+        run: scripts/protobuf_codegen.sh
+      - shell: bash
+        run: .github/workflows/check-clean-branch.sh
+  mock_gen:
+    name: Mocks
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: '~1.19.12'
+          check-latest: true
+      - run: go mod download
+      - run: go install github.com/golang/mock/mockgen@v1.6.0
```

Comment:
This is no longer maintained and we might want to use https://github.com/uber-go/mock instead (?). Not a strong opinion on this comment because we do use the google mock library currently.

Reply:
It wasn't being installed correctly when I didn't have this step: https://github.com/ava-labs/avalanchego/actions/runs/5811472964/job/15754782941?pr=1828

I think moving to uber-go/mock is outside of the scope of this PR since we'd need to re-gen all the mocks. FWIW I think we should switch off of mockgen

Reply:
Yeah it's outside of this PR's scope for sure. Just wanted to bring it up since we either need to switch to this repo eventually or figure out a new mock library/not use one.